pkg/kubectl/cmd/apply_edit_last_applied.go

```diff
@@ -0,0 +1,263 @@
+		kubectl apply edit-last-applied -f deploy.yaml -o json`)
+)
+
+func NewCmdApplyEditLastApplied(f cmdutil.Factory, out, err io.Writer) *cobra.Command {
+	options := &EditLastAppliedOptions{Out: out, ErrOut: err, Factory: f}
+	cmd := &cobra.Command{
+		Use:     "edit-last-applied (RESOURCE/NAME | -f FILENAME)",
+		Short:   "Edit latest last-applied-configuration annotations of a resource/object",
+		Long:    applyEditLastAppliedLong,
+		Example: applyEditLastAppliedExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(options.Complete(cmd, args))
+			cmdutil.CheckErr(options.Validate())
+			cmdutil.CheckErr(options.RunEditLastApplied())
+		},
+	}
+
+	usage := "to use to edit the resource"
+	cmd.Flags().BoolVar(&options.CreateAnnotation, "create-annotation", false, "Will create 'last-applied-configuration' annotations if current objects doesn't have one")
+	cmdutil.AddFilenameOptionFlags(cmd, &options.FilenameOptions, usage)
+	cmd.Flags().StringP("output", "o", "yaml", "Output format. One of: yaml|json.")
+	cmd.Flags().String("output-version", "", "DEPRECATED: To edit using a specific API version, fully-qualify the resource, version, and group (for example: 'jobs.v1.batch/myjob').")
+	cmd.Flags().MarkDeprecated("output-version", "editing is now done using the resource exactly as fetched from the API. To edit using a specific API version, fully-qualify the resource, version, and group (for example: 'jobs.v1.batch/myjob').")
+	cmd.Flags().MarkHidden("output-version")
+	cmd.Flags().Bool("windows-line-endings", gruntime.GOOS == "windows", "Use Windows line-endings (default Unix line-endings)")
+	cmdutil.AddRecordFlag(cmd)
+
+	return cmd
+}
+
+func (o *EditLastAppliedOptions) Complete(cmd *cobra.Command, args []string) error {
+	printer, err := getPrinter(cmd)
+	if err != nil {
+		return err
+	}
+	mapper, originalResult, updatedResultsGetter, cmdNamespace, err := getMapperAndResult(o.Factory, args, &o.FilenameOptions, NormalEditMode)
+	if err != nil {
+		return err
+	}
+	o.printer = printer.printer
+	o.ext = printer.ext
+	o.addHeader = printer.addHeader
+	o.updatedResultsGetter = updatedResultsGetter
+	o.Namespace = cmdNamespace
+	o.OutputFormat = cmdutil.GetFlagString(cmd, "output")
+	o.Codec = o.Factory.JSONEncoder()
+	o.Mapper = mapper
+	windowsLineEndings := cmdutil.GetFlagBool(cmd, "windows-line-endings")
+
+	infos, err := originalResult.Infos()
+	if err != nil {
+		return err
+	}
+
+	// generate the file to edit
+	o.EditBuffer = &bytes.Buffer{}
+	var w io.Writer = o.EditBuffer
+	if windowsLineEndings {
+		w = crlf.NewCRLFWriter(w)
+	}
+	if o.addHeader {
+		o.writeAnnotationsHeader(w)
+	}
+	l := &api.List{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "List",
+			APIVersion: "v1",
+		},
+	}
+
+	for ix := range infos {
+		info := infos[ix]
+		data, err := kubectl.GetOriginalConfiguration(info.Mapping, info.Object)
+		if err != nil {
+			return err
+		}
+		if data == nil && !o.CreateAnnotation {
```

Comment:
https://github.com/kubernetes/kubernetes/pull/41694#issuecomment-282030349, this is a flag add in `set-last-applied`, I thought `edit` should also support it ? I'm ok if we remove it.